control-plane-operator/main.go

```diff
@@ -266,8 +267,19 @@ func NewStartCommand() *cobra.Command {
 			}
 			return "", fmt.Errorf("couldn't locate operator container on deployment")
 		}
-		hostedClusterConfigOperatorImage, err = lookupOperatorImage(hostedClusterConfigOperatorImage)
-		if err != nil {
+
+		if err := wait.PollImmediate(5*time.Second, 30*time.Second, func() (bool, error) {
+			hostedClusterConfigOperatorImage, err = lookupOperatorImage(hostedClusterConfigOperatorImage)
+			if err != nil {
+				return false, err
+			}
+			// Apparently this is occasionally set to an empty string
+			if hostedClusterConfigOperatorImage == "" {
```

Comment:
Right now we break the retyring if we encounter an error because we assume that an error is fatal and not recoverable. The "We got an empty string back" is likely recoverable, so we retry.